server/channels/store/sqlstore/integrity.go

```diff
@@ -544,3 +547,61 @@ func CheckRelationalIntegrity(ss *SqlStore, results chan<- model.IntegrityCheckR
 	mlog.Info("Done with relational integrity checks")
 	close(results)
 }
+
+func GetAllValidUserIDsFromDMChannels(ss *SqlStore) ([]*model.User, error) {
+	records := []model.OrphanedRecord{}
+	dirtyIDs := []string{}
+	query, args, err := ss.getQueryBuilder().
+		Select().
+		Column("CT.ChannelName AS ParentId").
+		From("Channels AS CT").
+		Where(sq.Eq{"CT.Type": []model.ChannelType{model.ChannelTypeDirect}}).
+		ToSql()
+
+	if err != nil {
+		mlog.Error("There is an error with the DM Channel Name query")
+		return nil, err
+	}
+
+	err = ss.GetMasterX().Select(&records, query, args...)
+	if err != nil {
+		mlog.Error("there us an issue fetching the records for DM Channel Names")
+		return nil, err
+	}
+
+	for k := range records {
+		dirtyIDs = append(dirtyIDs, *records[k].ParentId)
+	}
+
+	userIDs := []string{}
+
+	for _, v := range dirtyIDs {
+		temp := strings.Split(v, "_")
+		userIDs = append(userIDs, temp[0])
+	}
+
+	for k := range userIDs {
+		userIDs[k] = strings.TrimPrefix(userIDs[k], "<")
+		userIDs[k] = strings.TrimSuffix(userIDs[k], ">")
+	}
```

Comment:
I don't understand this. DM channel names are of the format `userID__userID`. For example:  `uoao4pyo3jrpuxgdwn7f15btdc__ywxebfjhspgd7y5zqab3q1f81r`. Where does `<`, `>` come in this?

Reply:
I was following this format `<userID>__<userID>` verbatim and considered the `<` `>` as part of the potential string that was going to be returned.

Reply:
I would suggest to setup a local environment, create a few DM channels, see for yourself how the names look, and run some tests by manually editing the channel names and confirm that you get an error.